File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

char* solution(const char* cards1[], size_t cards1_len, const char* cards2[], size_t cards2_len, const char* goal[], size_t goal_len) {
    // 두 카드 뭉치의 인덱스 초기화
    size_t index1 = 0;
    size_t index2 = 0;

    // goal 배열을 순회하며 확인
    for (size_t i = 0; i < goal_len; i++) {
        // 현재 goal 단어가 cards1의 다음 단어와 일치하면 cards1 인덱스 증가
        if (index1 < cards1_len && strcmp(goal[i], cards1[index1]) == 0) {
            index1++;
        }
        // 현재 goal 단어가 cards2의 다음 단어와 일치하면 cards2 인덱스 증가
        else if (index2 < cards2_len && strcmp(goal[i], cards2[index2]) == 0) {
            index2++;
        }
        // 둘 다 일치하지 않으면 만들 수 없음
        else {
            char* answer = (char*)malloc(3);
            strcpy(answer, "No");
            return answer;
        }
    }

    // 모든 단어를 성공적으로 사용하여 goal을 만들 수 있으면 "Yes" 반환
    char* answer = (char*)malloc(4);
    strcpy(answer, "Yes");
    return answer;
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
public class Solution {
    public static String solution(String[] cards1, String[] cards2, String[] goal) {
        int index1 = 0; // cards1의 현재 위치
        int index2 = 0; // cards2의 현재 위치

        for (String word : goal) {
            // cards1의 현재 단어와 goal 단어가 일치하면 index1 증가
            if (index1 < cards1.length && word.equals(cards1[index1])) {
                index1++;
            }
            // cards2의 현재 단어와 goal 단어가 일치하면 index2 증가
            else if (index2 < cards2.length && word.equals(cards2[index2])) {
                index2++;
            }
            // 둘 다 일치하지 않으면 goal을 구성할 수 없음
            else {
                return "No";
            }
        }

        // 모든 단어를 성공적으로 사용하여 goal을 만들 수 있으면 "Yes" 반환
        return "Yes";
    }
}
